src/common/transformations/src/transformations/fp16_compression/mark_decompression_convert_constant_folding.cpp

```diff
@@ -74,3 +76,35 @@ pass::KeepConstAndDecompression::KeepConstAndDecompression() {
     auto m = std::make_shared<pattern::Matcher>(node_pattern, matcher_name);
     register_matcher(m, callback);
 }
+
+pass::KeepConstantsPrecisionAndAddConverts::KeepConstantsPrecisionAndAddConverts() {
+    MATCHER_SCOPE(KeepConstantsPrecisionAndAddConverts);
+    auto const_pattern = pattern::wrap_type<ov::op::v0::Constant>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto const_node = m.get_match_root();
+
+        if (transformation_callback(const_node)) {
+            return false;
+        }
+
+        auto constant_target_inputs = const_node->get_output_target_inputs(0);
+        auto convert = std::make_shared<ov::op::v0::Convert>(const_node, const_node->get_element_type());
```

Comment:
Seems like you always insert this dummy `Convert` op for matmuls, so I wonder what happens in case of FP16 model being loaded with FP16 inference precision (i.e. when conversion is not needed)? Is this convert eliminated on early transformation stages (in common optimizations for example)?

Reply:
Yes, looks like this Convert must be removed in CommonOptimizations->MOCTransformations->SharedOpOptimization pass